symfony/client/src/orangehrmPimPlugin/pages/employee/EmployeeJob.vue

```diff
@@ -0,0 +1,303 @@
+import EditEmployeeLayout from '@/orangehrmPimPlugin/components/EditEmployeeLayout';
+import SwitchInput from '@orangehrm/oxd/core/components/Input/SwitchInput';
+import SubunitDropdown from '@/orangehrmPimPlugin/components/SubunitDropdown';
+import EmploymentStatusDropdown from '@/orangehrmPimPlugin/components/EmploymentStatusDropdown';
+
+const jobDetailsModel = {
+  joinedDate: [],
+  jobTitleId: [],
+  empStatusId: [],
+  jobCategoryId: [],
+  subunitId: [],
+  locationId: [],
+};
+
+const contractDetailsModel = {
+  startDate: '',
+  endDate: '',
+  contractAttachment: null,
+};
+
+export default {
+  components: {
+    'edit-employee-layout': EditEmployeeLayout,
+    'oxd-switch-input': SwitchInput,
+    'subunit-dropdown': SubunitDropdown,
+    'employment-status-dropdown': EmploymentStatusDropdown,
+  },
+
+  props: {
+    empNumber: {
+      type: String,
+      required: true,
+    },
+    locations: {
+      type: Array,
+      default: () => [],
+    },
+    jobTitles: {
+      type: Array,
+      default: () => [],
+    },
+    jobCategories: {
+      type: Array,
+      default: () => [],
+    },
+    subunits: {
+      type: Array,
+      default: () => [],
+    },
+    employmentStatuses: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  setup(props) {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      `api/v2/pim/employees/${props.empNumber}/job-details`,
+    );
+
+    return {
+      http,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      createContractDetails: false,
+      job: {...jobDetailsModel},
+      contract: {...contractDetailsModel},
+      rules: {
+        joinedDate: [
+          v => {
+            return !v || v?.length <= 100 || 'Should not exceed 100 characters';
+          },
+        ],
+        endDate: [
+          v => {
+            return !v || v?.length <= 100 || 'Should not exceed 100 characters';
+          },
+        ],
+        
+      },
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .request({
+          method: 'PUT',
+          data: {
+            ...this.job,
+            jobTitleId: this.job.jobTitleId.map(item => item.id)[0],
+            jobCategoryId: this.job.jobCategoryId.map(item => item.id)[0],
+            subunitId: this.job.subunitId.map(item => item.id)[0],
+            empStatusId: this.job.empStatusId.map(item => item.id)[0],
+            locationId: this.job.locationId.map(item => item.id)[0],
+          },
+        })
+        .then(response => {
+          this.updateJobModel(response);
+          if (this.createContractDetails) {
+            return this.http.request({
+              method: 'PUT',
+              url: `api/v2/pim/employees/${this.empNumber}/employment-contract`,
+              data: {
+                ...this.contract,
+                startDate: this.contract.startDate,
```

Comment:
Here manually setting the fields are not needed since `this.contract` contains it.